services/paymaster/src/db/config.rs

```diff
@@ -0,0 +1,177 @@
+use crate::config_manager::config::{Config, Domain};
+use crate::constraint::TransactionVariation;
+use crate::db::pool::pool;
+use std::collections::HashMap;
+use url::Url;
+
+use argon2::{
+    password_hash::{rand_core::OsRng, PasswordHasher, SaltString},
+    Argon2,
+};
+use sqlx::{types::Json, FromRow};
+use uuid::Uuid;
+
+use crate::db::pool;
+
+/// A row from the database that represents a domain and its transaction variations.
+#[derive(FromRow)]
+struct Row {
+    domain_id: Uuid,
+    domain: String,
+    enable_session_management: bool,
+    enable_preflight_simulation: bool,
+    instructions: Json<TransactionVariation>,
+}
+
+/// Load the config from the database
+pub async fn load_config() -> Result<Config, sqlx::Error> {
+    let rows: Vec<Row> = sqlx::query_as::<_, Row>(
+        r#"
+        SELECT
+          d.id            AS domain_id,
+          d.domain        AS domain,
+          d.enable_session_management,
+          d.enable_preflight_simulation,
+          v.instructions  AS instructions
+        FROM domain_config d
+        JOIN variation v ON v.domain_config_id = d.id
+        ORDER BY d.id
+        "#,
+    )
+    .fetch_all(pool::pool())
+    .await?;
+
+    let mut map: HashMap<Uuid, Domain> = HashMap::new();
+
+    for r in rows {
+        map.entry(r.domain_id)
+            .or_insert_with(|| Domain {
+                domain: r.domain,
+                enable_session_management: r.enable_session_management,
+                enable_preflight_simulation: r.enable_preflight_simulation,
+                tx_variations: Vec::new(),
+            })
+            .tx_variations
+            .push(r.instructions.0);
+    }
+
+    Ok(Config {
+        domains: map.into_values().collect(),
+    })
+}
+
+/// Hash a plaintext password using Argon2.
+fn hash_password(plain: &str) -> String {
+    let salt = SaltString::generate(&mut OsRng);
+
+    // Argon2 with default params (Argon2id v19)
+    let argon2 = Argon2::default();
+
+    // Hash password to PHC string ($argon2id$v=19$...)
+    argon2
+        .hash_password(plain.as_bytes(), &salt)
+        .unwrap()
```

Comment:
see note about unwrap(), better practice is to return a Result for `hash_password`